CONTRIBUTING.md

```diff
@@ -77,6 +78,64 @@ to do this is to invoke `rustc` like this:
 $ RUST_BACKTRACE=1 rustc ...
 ```
 
+## The Build System
+
+The build system for Rust is complex. It covers bootstrapping the compiler,
+running tests, building documentation and more.
+
+If Makefiles are your thing, all the configuration lives in
+[the `mk` directory][mkdir] in the project root. Is can be hard to follow
+in places, as it uses some advanced Make features which make for some
+challenging reading. Running `make help` and `make tips` might be useful when
```

Comment:
This document already contains most of the information of `make help`. I would point towards `make nitty-gritty` but I have no idea if it's even remotely up to date. Perhaps cut the false reassurance of documentation and just point people towards `#rust-internals`.